{{ cookiecutter.package_name }}/.github/workflows/ci.yml

```diff
@@ -0,0 +1,98 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - 'main'
+      - '*.*'
+      - '!*backport*'
+    tags:
+      - 'v*'
+      - '!*dev*'
+      - '!*pre*'
+      - '!*post*'
+  pull_request:
+  # Allow manual runs through the web UI
+  workflow_dispatch:
+
+concurrency:
+  group: {{ '${{ github.workflow }}-${{ github.ref }}' }}
+  cancel-in-progress: true
+
+jobs:
+  core:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      envs: |
+        - linux: py312
+    secrets:
+      CODECOV_TOKEN: {{ '${{ secrets.CODECOV_TOKEN }}' }}
+
+  sdist_verify:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - run: python -m pip install -U --user build
+      - run: python -m build . --sdist
+      - run: python -m pip install -U --user twine
+      - run: python -m twine check dist/*
+
+  test:
+    needs: [core, sdist_verify]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto
+      envs: |
+        - windows: py311
+        - macos: py310
+        - linux: py310-oldestdeps
+    secrets:
+      CODECOV_TOKEN: {{ '${{ secrets.CODECOV_TOKEN }}' }}
+
+  docs:
+    needs: [core]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      default_python: '3.10'
+      submodules: false
+      pytest: false
+      toxdeps: tox-pypi-filter
+      envs: |
+        - linux: build_docs
+
+  cron:
+    if: |
+      github.event_name == 'workflow_dispatch' || (
+        github.event_name == 'pull_request' &&
+        contains(github.event.pull_request.labels.*.name, 'Run cron CI')
+      )
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      default_python: '3.10'
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      envs: |
+        - linux: py311-devdeps
+    secrets:
+      CODECOV_TOKEN: {{ '${{ secrets.CODECOV_TOKEN }}' }}
+
+  publish_pure:
+    needs: [test]
```

Comment:
We don't block on docs in core since it can be flaky with the examples, but maybe we should here?

Reply:
The non-gallery core build shouldn't be flaky no? Yeah I think that's a good idea anyway, you shouldn't release if your docs are busted.

Reply:
Oh yeah, that isnt, we should block on that at least in future.